### ISSUE TITLE
Fails to start 

### ISSUE THREAD

**Issue by danielsteiner** [spantaleev/matrix-docker-ansible-deploy#961]
Hi, i just tried to run this ansible book and it worked fine at first but after a restart, things started to be messed up which was why I tried to uninstall and install it again now I get this error at setup-all / start.

`TASK [matrix-postgres : Execute Postgres additional database initialization SQL file for synapse] *************************************************************************************************************
fatal: [matrix.removed.com]: FAILED! => {"changed": true, "cmd": ["/usr/bin/env", "docker", "run", "--rm", "--user=997:1002", "--cap-drop=ALL", "--env-file=/matrix/postgres/env-postgres-psql", "--network", "matrix", "--mount", "type=bind,src=/tmp/matrix-postgres-init-additional-db-user-and-role.sql,dst=/matrix-postgres-init-additional-db-user-and-role.sql,ro", "--entrypoint=/bin/sh", "docker.io/postgres:13.2-alpine", "-c", "psql -h matrix-postgres --file=/matrix-postgres-init-additional-db-user-and-role.sql"], "delta": "0:00:00.045951", "end": "2021-03-24 03:34:59.763153", "msg": "non-zero return code", "rc": 125, "start": "2021-03-24 03:34:59.717202", "stderr": "docker: Error response from daemon: invalid mount config for type \"bind\": bind source path does not exist: /tmp/matrix-postgres-init-additional-db-user-and-role.sql.\nSee 'docker run --help'.", "stderr_lines": ["docker: Error response from daemon: invalid mount config for type \"bind\": bind source path does not exist: /tmp/matrix-postgres-init-additional-db-user-and-role.sql.", "See 'docker run --help'."], "stdout": "", "stdout_lines": []}`

This worked fine in the first run. 

I also removed all images / containers from my system. 

Any idea what to do?

**Comment by aaronraimist** [comment]
What do you mean "things started to be messed up"? How did you uninstall? Did you follow https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/uninstalling.md?

**Comment by danielsteiner** [comment]
Hi, 

i wasn’t able to start any _other_ docker containers anymore (my Webserver / reverse proxy e.g.) because of used ports (they weren‘t used) 

Uninstalled as in „ ansible-playbook -i inventory/hosts setup.yml --tags=uninstall“ and yes, of course i followed he uninstalling docs.

**Comment by aaronraimist** [comment]
I'm not sure. Make sure you are on the latest commit and then maybe try rebooting the server. Did the `Create additional database initialization SQL file for synapse` task just above that task work?

**Comment by ksokhan** [comment]
Also seeing this error. System was working fine until a recent update to latest master, and reinstallation. I tried reisntalling, and also removing `/matrix` directory and rerunning install to no avail. 

Additionally, I discovered matrix-postgres service fails on start with:
`docker: Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /matrix/postgres/data` in my syslog. So wondering if that could be part of the problem?

**Comment by danielsteiner** [comment]
I just manually uninstalled matrix again (as described in docs/uninstalling.md), pulled the latest version from github but still get the same error.

**Comment by Devenda** [comment]
Same issue here, however in my case it seems it was not linked to a git pull.
After a reboot my Facebook bridge no longer worked and I tried rerunning the install script but got this error.

After that I did a pull, but the error remains.

I even had a similar error (maybe even the same) a month ago and the only 'solution' was a full reinstall on a fresh server.

**Comment by Devenda** [comment]
So, I tried a few things, read some logfiles, removed a bunch of stuff and finally got it working again.

In the logfiles I found this message:

`Apr  1 21:07:03 matrix docker.dockerd[626]: time="2021-04-01T21:07:03.937880247Z" level=error msg="ee46aa213e9479dbc971212c87f77022b2bc22f845e9e41b43f63592efad4736 cleanup: failed to delete container from containerd: no such container"
Apr  1 21:07:03 matrix docker.dockerd[626]: time="2021-04-01T21:07:03.991932021Z" level=error msg="Handler for POST /v1.40/containers/ee46aa213e9479dbc971212c87f77022b2bc22f845e9e41b43f63592efad4736/start returned error: error while creating mount source path '/matrix/mautrix-facebook/config': mkdir /matrix: read-only file system"
Apr  1 21:07:03 matrix matrix-mautrix-facebook[15468]: docker: Error response from daemon: error while creating mount source path '/matrix/mautrix-facebook/config': mkdir /matrix: read-only file system.
Apr  1 21:07:03 matrix matrix-mautrix-facebook[15468]: time="2021-04-01T21:07:03Z" level=error msg="error waiting for container: context canceled"`

This (and the error posted by the OP) sounded like permission issues.

This brought me to: https://stackoverflow.com/questions/45764477/docker-compose-error-while-creating-mount-source-path
(do note this is all on Ubuntu 20.04.2 LTS).

Removal worked but reinstalling using the guide in the docker link did not, after removing using these steps: 
https://docs.docker.com/engine/install/ubuntu/#uninstall-docker-engine

manually renaming docker.list to docker.list.old in /etc/apt/sources.list.d

and starting the daemon using
`sudo systemctl start docker`

I was able to re run the installer script successfully.

As the script seems to install docker itself it may just have been the removal of docker that did the trick.

Note: Contrary to what I expected no data was lost (even after removing all of docker and its folders (eg rm -R /var/lib/docker)) but please be mindful that this all me result in data loss...

**Comment by spantaleev** [comment]
Sounds like a borked up Docker install. Deleting `/var/lib/docker` is fine, as far as the playbook is concerned. It only stores things in `/matrix`.

**Comment by danielsteiner** [comment]
Well some people might have had docker running _before_ they tried this playbook so its not really fine to „just delete“ `/var/lib/docker`

**Comment by Devenda** [comment]
> Well some people might have had docker running _before_ they tried this playbook so its not really fine to „just delete“ `/var/lib/docker`

Indeed, even then I am not sure that it was this that solved the issue.
In my case the server is dedicated to matrix so nothing else should have interfered with the install.

**Comment by smarti57** [comment]
Same thing here... used to work.. no longer works.  I've been trying to keep my setup since it was flawless prior to update, but I've spent quite a bit of time so far trying to unstick it's current state.  It looks like the exact same issue.

```
  stderr: |-
    docker: Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /tmp/matrix-postgres-init-additional-db-user-and-role.sql.
```

What's really odd, is that it's gotten very hard to run any of the other containers.  I've seen some comments here about docker being the issue, but the only thing on this machine is this SW, I think it's a bug in sequence of updates and if for some reason an update doesn't happen it gets wedged.

**Comment by xXTim97Xx** [comment]
I have the same problem here. Does somebody have a solution for this?
My file /tmp/matrix-postgres-init-additional-db-user-and-role.sql. does exist and the rights seems to be correct

**Comment by Devenda** [comment]
I have yet to have the same issue since posting the above comment.
However I have recently rerun the installer script with zero issues...

**Comment by smarti57** [comment]
I ended up reinstalling after backing up db.  Everything has been smooth sailing since then.
